lambda/events_id/delete/index.js

```diff
@@ -1,96 +1,97 @@
+    console.log("ak-api-travis-lambda lambda function started...");
+    
+    // this stores the input given by the caller, a UUID (uuid v1) for a specific item in the ak-api-travis-dynamodb table
+    const ID = event.params.path['event_id'];
+    
+    // log for recording the given url parameter given
+    console.log("Path parameter, event_id: ", ID);
+    
 	// this stores the http method used by the caller and is needed when determining whether the call is a 
 	// DELETE, PUT, or GET request
-	const httpMethod = event.httpMethod;
-
-	// validate input parameter as uuid version 1, when fails a 400 response must be given
-	if (!validate(ID, 1)) {
-		var response = {
-			statusCode: 400,
-			headers: {
-				"Access-Control-Allow-Origin": "*"
-			},
-		}
-		return response;
-	}
-
-
-	// determine what http request is made to handle it appropriately 
-	// *** only using DELETE now but will need the others soon
-	if (httpMethod === 'DELETE') {
-
-		// params contains the table name and primary key to delete an item with the given url parameter
-		var params = {
-			TableName: "ak-api-travis-dynamodb",
-			Key: {
-				"event_id": ID
-			},
-			// needed to return deleted item or empty object if nothing found to delete by given ID
-			"ReturnValues": "ALL_OLD"
-		};
-
-		let code;
-		try {
-			// delete method is needed to delete items from a Dynamodb table
-			const data = await documentClient.delete(params).promise();
-
-			// check if nothing was returned to cause a 404 response, item not found in db 
-			if (Object.entries(data).length === 0 && data.constructor === Object) {
-				code = 404;
-				// log needed for when a parameter eventID is not found in the db
-				console.log(`eventID ${ID} not found in Table.`);
-			}
-			else {
-				code = 200;
-				// log needed to confirm successful deletion of item in db
-				console.log("Item successfully deleted.");
+    //const httpMethod = event.httpMethod;
+    const httpMethod = event.context['http-method'];
+    
+    //Log calling API type and endpoint
+    console.log("API-Endpoint called: " + httpMethod + " " + event.context['resource-path']);
+    
+    //Log calling IP
+    console.log("Calling-IP: " + event.params.header['X-Forwarded-For']);
+    
+    // status response needed for API Gateway to return the correct response
+    const response = {
+		statusCode: 200
+	};
+    
+    
+    // validate input parameter as uuid version 1, when fails a 400 response must be given
+    if(!validate(ID, 4)) {
+    	console.log("ERROR: uuid in not uuid/v4");
+    	response.statusCode = 400;
+		callback(JSON.stringify(response), null);
+	
+    }
+    else {
+	
+	   // determine what http request is made to handle it appropriately 
+	   // *** only using DELETE now but will need the others soon
+	   if (httpMethod === 'DELETE') {
+	   	
+			// params contains the table name and primary key to delete an item with the given url parameter
+		    var params = {
+		        TableName : "ak-api-dynamo",
```

Comment:
The TableName should be changed to ak-prod-events-dynamo to work with the prod environment.

Reply:
Approve the change